viewing_party/party.py

```diff
@@ -1,23 +1,193 @@
+
+    seen_titles = set()
+
+    for friend in friends:
+        for movie in friend["watched"]:
+            title = movie["title"]
+
+            # If the user hasn't watched it and it's not already added
+            if title not in user_titles and title not in seen_titles:
+                friends_unique.append(movie)
+                # Title that seen to avoid duplicates
+                seen_titles.add(title)
+
+    #list that friends watched
+    return friends_unique
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+# method 1 
+def get_available_recs(user_data):
+    # Step 1: Get friends' unique watched movies using Wave 3
+    friends_unique_watched = get_friends_unique_watched(user_data)
+
+    # Step 2: From these movies, filter out those hosted on platforms the user subscribes to
+    subscriptions = user_data["subscriptions"]
+    recommended_movies = []
+
+    for movie in friends_unique_watched:
+        if movie["host"] in subscriptions:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
+# method 2
+def get_available_recs(user_data):
+    # Step 1: Get a list of movies that friends have watched, with no duplicate titles
+    friend_watched_list = set()
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_watched_list.add(movie["title"])
+    
+    # Step 2: Filter out movies that the user has already watched
+    user_watched_list = set()
+    for movie in user_data["watched"]:
+        user_watched_list.add(movie["title"])
+
+    friend_watched_user_not_list = []
+    for title in friend_watched_list:
+        if title not in user_watched_list:
+            friend_watched_user_not_list.append(title)
+
+    # Step 3: From the remaining movies, find those hosted on platforms the user subscribes to
+    movie_in_host = []
+    for friend in user_data ["friends"]:
+        for movie in friend["watched"]:
+            if (movie["title"] in friend_watched_user_not_list 
+                and movie["host"] in user_data["subscriptions"] 
+                and movie not in movie_in_host):
+                movie_in_host.append(movie)
+    return movie_in_host
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+
+    most_frequent_genre = get_most_watched_genre(user_data)
+
+    #Determine a list of recommended movies. 
+    recommended_movies = []
+
+    #List of titles the user has already watched
+    user_watched_titles = [movie["title"] for movie in user_data["watched"]]
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if (movie["title"] not in user_watched_titles and
+                movie["genre"] == most_frequent_genre and
+                movie not in recommended_movies):

```

Comment:
While you break this expression up over 3 lines, it's still a little challenging to digest because there's a lot of logic crammed in here. How could you refactor this method to make it more readable and concise?

You already wrote a function that returns a list of movies a user's friends have watched but the user hasn't watched. You should invoke that here instead of writing additional logic.